client/state/signup/steps/website-content/reducer.ts

```diff
@@ -20,16 +20,15 @@ export default withSchemaValidation(
 					websiteContent: action.payload,
 				};
 			case SIGNUP_STEPS_WEBSITE_CONTENT_INITIALIZE_PAGES: {
-				// Initialize with provided pages only if given pages do not exist
 				// When initializing page we need to leave pages that already exist alone
 				// so that persisted state can load correctly
-				const existingPageIDs = state.websiteContent.map( ( page ) => page.id );
-				const newPages = action.payload.filter(
-					( page: PageData ) => ! existingPageIDs.includes( page.id )
-				);
+				const newPages = action.payload.map( ( page: PageData ) => ( {
```

Comment:
This change fixes this bug:
1. Go to `/start/site-content-collection/website-content?siteSlug=<slug1>`. Confirm that there are 4 pages.
2. Go to `/start/site-content-collection/website-content?siteSlug=<slug2>`. `slug2` should be a website with a different category. For example, if `slug1` is "local-services", then `slug2` can be "restaurant". On visiting this URL, 5 pages are visible.

The number of pages shown should be fixed and should always be initialized [here](https://github.com/Automattic/wp-calypso/pull/60998/files#diff-16e080d2ad3a8e19863abe08ffd798f9be323293b0171c0e6f5efdaa7dedd59fR68).

Reply:
Nice catch! :)